commit 4ea6a8046bb49d43c950898f0cb4e1994ef6c89d
tree cf7ab6f3a88ef7c66f28322febe18d7694ab27b9
parent 444d1d9bb5b724f03344c9317bc01d54a9b39073
author Yan Zheng <yanzheng@21cn.com> Mon, 24 Oct 2005 19:55:23 +0800
committer Arnaldo Carvalho de Melo <acme@mandriva.com> Tue, 25 Oct 2005 21:17:52 -0200

    [IPV6]: Fix refcnt of struct ip6_flowlabel
    
    Signed-off-by: Yan Zheng <yanzheng@21cn.com>
    Acked-by: YOSHIFUJI Hideaki <yoshfuji@linux-ipv6.org>
    Signed-off-by: Arnaldo Carvalho de Melo <acme@mandriva.com>

diff --git a/net/ipv6/ip6_flowlabel.c b/net/ipv6/ip6_flowlabel.c
index f841bde..bbbe80c 100644
--- a/net/ipv6/ip6_flowlabel.c
+++ b/net/ipv6/ip6_flowlabel.c
@@ -483,7 +483,7 @@ int ipv6_flowlabel_opt(struct sock *sk, 
 						goto done;
 					}
 					fl1 = sfl->fl;
-					atomic_inc(&fl->users);
+					atomic_inc(&fl1->users);
 					break;
 				}
 			}
